plugins/Feedback/Feedback.php

```diff
@@ -64,17 +68,73 @@ public function getClientSideTranslationKeys(&$translationKeys)
         $translationKeys[] = 'Feedback_PleaseLeaveExternalReviewForMatomo';
         $translationKeys[] = 'Feedback_RemindMeLater';
         $translationKeys[] = 'Feedback_NeverAskMeAgain';
+        $translationKeys[] = 'Feedback_ReferMatomo';
+        $translationKeys[] = 'Feedback_ReferBannerTitle';
+        $translationKeys[] = 'Feedback_ReferBannerLonger';
+        $translationKeys[] = 'Feedback_ReferBannerSocialShareText';
+        $translationKeys[] = 'Feedback_ReferBannerEmailShareSubject';
+        $translationKeys[] = 'Feedback_ReferBannerEmailShareBody';
         $translationKeys[] = 'General_Ok';
         $translationKeys[] = 'General_Cancel';
     }
 
-    public function renderFeedbackPopup(&$pageHtml)
+    public function renderViewsAndAddToPage(&$pageHtml)
+    {
+        $feedbackPopopView = $this->renderFeedbackPopup();
+        $referBannerView = $this->renderReferBanner();
+
+        $views = [$feedbackPopopView, $referBannerView];
+        $implodedViews = implode('', $views);
+
+        $endOfBody = strpos($pageHtml, '</body>');
+        $pageHtml = substr_replace($pageHtml, $implodedViews, $endOfBody, 0);
+    }
+
+    public function renderFeedbackPopup()
     {
         $popupView = new View('@Feedback/feedbackPopup');
         $popupView->promptForFeedback = (int)$this->getShouldPromptForFeedback();
-        $popupHtml = $popupView->render();
-        $endOfBody = strpos($pageHtml, "</body>");
-        $pageHtml = substr_replace($pageHtml, $popupHtml, $endOfBody, 0);
+
+        return $popupView->render();
+    }
+
+    public function renderReferBanner()
+    {
+        $referBannerView = new View('@Feedback/referBanner');
+        $referBannerView->showReferBanner = (int) $this->showReferBanner();
+
+        return $referBannerView->render();
+    }
+
+    public function showReferBanner()
+    {
+        if (Piwik::isUserIsAnonymous()) {
+            return false;
+        }
+
+        if (!Piwik::hasUserSuperUserAccess()) {
+            return false;
+        }
+
+        if ($this->isDisabledInTestMode()) {
+            return false;
+        }
+
```

Comment:
Here, can we add an `@internal` event like `Feedback.showReferBanner` that takes a bool reference:

```
$shouldShowReferBanner = true;

/**
 * @internal
 */
Piwik::postEvent('Feedback.showReferBanner', [&$shouldShowReferBanner]);

if (!$shouldShowReferBanner) {
    return false;
}
```

This way, the WhiteLabel plugin can hook into it and make sure it's not shown at all when the plugin is active.